src/engine/x86-64/X86_64Interpreter.v3

```diff
@@ -2220,6 +2220,43 @@ class X86_64InterpreterGen(ic: X86_64InterpreterCode, w: DataWriter) {
 		asm_mov_m_r(mem_addr, val);
 		endHandler();
 	}
+	def genV128Splat(opcode: Opcode, from_mem: bool,
```

Comment:
I think this will be cleaner if you either split it to two functions (one from memory and one not). Further, it will be simpler to also just pass the `MacroAssembler` method for the actual splat, rather than switching again on the opcode.

Reply:
I refactored `genSplat` to take in a `gen` method as an additional argument to avoid switching on the opcode.